File: frontend/src/features/auth/services/authService.js
import apiClient from '../../core/api/apiClient';
import { API_ENDPOINTS } from '../../shared/utils/constants';

// Auth service functions
export const authService = {
  login: async (credentials) => {
    const response = await apiClient.post(API_ENDPOINTS.AUTH.LOGIN, credentials);
    return response.data;
  },

  register: async (userData) => {
    const response = await apiClient.post(API_ENDPOINTS.AUTH.REGISTER, userData);
    return response.data;
  },

  logout: async () => {
    const response = await apiClient.post(API_ENDPOINTS.AUTH.LOGOUT);
    return response.data;
  },

  refreshToken: async (refreshToken) => {
    const response = await apiClient.post(API_ENDPOINTS.AUTH.REFRESH, { refreshToken });
    return response.data;
  },

  getCurrentUser: async () => {
    const response = await apiClient.get(API_ENDPOINTS.USERS.ME);
    return response.data;
  }
};

export default authService;

File: backend/src/main/java/com/myproject/nexa/services/AuthService.java
package com.myproject.nexa.services;

import com.myproject.nexa.dto.request.LoginRequest;
import com.myproject.nexa.dto.request.RegisterRequest;
import com.myproject.nexa.dto.request.RefreshTokenRequest;
import com.myproject.nexa.dto.response.AuthResponse;

public interface AuthService {
    AuthResponse login(LoginRequest loginRequest);
    AuthResponse register(RegisterRequest registerRequest);
    AuthResponse refreshToken(RefreshTokenRequest refreshTokenRequest);
    void logout(String refreshToken);
}